Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -26,8 +26,20 @@
     return d2dFactory;
 }
 
-// Helper for getting a D2DFactory
 ComPtr<ID2D1Factory> _GetD2DFactoryInstance() {
     static ComPtr<ID2D1Factory> factory = __createD2DFactory();
     return factory;
+}
+
+static ComPtr<IWICImagingFactory> __createWICFactory() {
+  ComPtr<IWICImagingFactory> wicFactory;
+  THROW_IF_FAILED(CoCreateInstance(CLSID_WICImagingFactory, nullptr,
```

Comment:
seems like a better pattern is to maybe return nullptr for failures and then have callers check that and do the right thing (return a nullptr CGImageRef or whatever)

---
In reply to: [84810633](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84810633) [](ancestors = 84810633)